api/access/server.go

```diff
@@ -0,0 +1,515 @@
+		Height: height,
+	}
+
+	return s.GetBlockByHeight(ctx, req)
+}
+
+func (s *Server) GetBlockByID(ctx context.Context, in *access.GetBlockByIDRequest) (*access.BlockResponse, error) {
+	blockID := flow.HashToID(in.Id)
+	height, err := s.index.HeightForBlock(blockID)
+	if err != nil {
+		return nil, fmt.Errorf("could not get height for block %x: %w", blockID, err)
+	}
+
+	req := access.GetBlockByHeightRequest{
+		Height: height,
+	}
+
+	return s.GetBlockByHeight(ctx, &req)
+}
+
+func (s *Server) GetBlockByHeight(_ context.Context, in *access.GetBlockByHeightRequest) (*access.BlockResponse, error) {
+	header, err := s.index.Header(in.Height)
+	if err != nil {
+		return nil, fmt.Errorf("could not get header for height %d: %w", in.Height, err)
+	}
+
+	sealIDs, err := s.index.SealsByHeight(in.Height)
+	if err != nil {
+		return nil, fmt.Errorf("could not get seals for height %d: %w", in.Height, err)
+	}
+
+	seals := make([]*entities.BlockSeal, 0, len(sealIDs))
+	for _, sealID := range sealIDs {
+		seal, err := s.index.Seal(sealID)
+		if err != nil {
+			return nil, fmt.Errorf("could not get seal with ID %x: %w", sealID, err)
+		}
+
+		blockID := seal.BlockID
+		resultID := seal.ResultID
+
+		// See https://github.com/onflow/flow-go/blob/v0.17.4/engine/common/rpc/convert/convert.go#L180-L188
+		entity := entities.BlockSeal{
+			BlockId:                    blockID[:],
+			ExecutionReceiptId:         resultID[:],
+			ExecutionReceiptSignatures: [][]byte{}, // filling seals signature with zero
+		}
+		seals = append(seals, &entity)
+	}
+
+	collIDs, err := s.index.CollectionsByHeight(in.Height)
+	if err != nil {
+		return nil, fmt.Errorf("could not get collections for height %d: %w", in.Height, err)
+	}
+
+	collections := make([]*entities.CollectionGuarantee, 0, len(collIDs))
+	for _, collID := range collIDs {
+		guarantee, err := s.index.Guarantee(collID)
+		if err != nil {
+			return nil, fmt.Errorf("could not get collection with ID %x: %w", collID, err)
+		}
+
+		entity := entities.CollectionGuarantee{
+			CollectionId: conv.IDToHash(collID),
+			Signatures:   [][]byte{guarantee.Signature},
+		}
+		collections = append(collections, &entity)
+	}
+
+	blockID := header.ID()
+	block := entities.Block{
+		Id:                   blockID[:],
+		Height:               in.Height,
+		ParentId:             header.ParentID[:],
+		Timestamp:            timestamppb.New(header.Timestamp),
+		CollectionGuarantees: collections,
+		BlockSeals:           seals,
+		Signatures:           [][]byte{header.ParentVoterSig},
+	}
+
+	resp := access.BlockResponse{
+		Block: &block,
+	}
+
+	return &resp, nil
+}
+
+func (s *Server) GetCollectionByID(_ context.Context, in *access.GetCollectionByIDRequest) (*access.CollectionResponse, error) {
+	collId := flow.HashToID(in.Id)
+	collection, err := s.index.Collection(collId)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve collection with ID %x: %w", in.Id, err)
+	}
+
+	collEntity := entities.Collection{
```

Comment:
Also inconsistent naming, should then all be `entity`, but prefer the `msg` suffix.